### ISSUE TITLE
ForgeUI - Snaping windows to grid

### ISSUE THREAD

**Issue by adamjedlicka** [ForgeUI/ForgeUI#44]

**Comment by Toludin** [comment]
The way I'd like to do this (snap when you release left click) is currently blocked by the MouseButtonUp event not firing on left click up. I'm looking into alternatives, but it might need to wait for that bug to be fixed.

**Comment by mkaito** [comment]
Wouldn't that be counter intuitive? Most snap-to-grid implementations have the element move in increments while mouse down and things stay in place when you release the button. Having things move after releasing the  button might be confusing.

**Comment by Toludin** [comment]
Hmm, I'll look into that possibility. That would certainly get around the current API limitation. 'Snap to Grid' made me think of a release-and-snap method off the top of my head, but a locking movement to the grid itself would certainly be useful.

**Comment by mkaito** [comment]
That's how I remember other snap to grid implementations off the top of my head. Might be worth looking into.

While you're looking at snapping to grid, would snapping to other windows also be a possibility?

**Comment by Toludin** [comment]
Yea if we can do the former, I don't see why the latter would be much harder to implement. Need to try it out tonight.

**Comment by adamjedlicka** [comment]
Both snap methoths are possible to do in WS right now. Snaping to grid shouldn't be that hard, but snaping to other windows? Geez. I'm not doing that :D

BTW @Toludin I will have some time tommorow so I'll fix some things and release new version.

**Comment by Toludin** [comment]
Thanks :-D

**Comment by Toludin** [comment]
Pretty close to done with this, though I haven't moved it forward much pending that release. Is that still coming, Adam?

**Comment by jcdenton2k** [comment]
I'd love to see this done as well.  I was about to open a new issue but I'm glad I found this (relatively recent comment).
